### ISSUE TITLE
Gradient disappearance occurs during training

### ISSUE THREAD

**Issue by Xinghetayue** [declare-lab/Multimodal-Infomax#20]
The following error occurs when executing "'python main.py --dataset mosei --contrast". It seems that there is a problem during the training process that causes NaN values to appear in the gradient calculation. I would like to know how to solve it.
![1](https://github.com/declare-lab/Multimodal-Infomax/assets/79571907/cb298956-83dc-4f18-b74c-f96193b87cc4)

**Comment by Clement25** [comment]
Hi, could you please check if you were running with the torch version in 'requirement.txt'?

**Comment by Xinghetayue** [comment]
> Hi, could you please check if you were running with the torch version in 'requirement.txt'?

Thanks for the reply. My torch version is 1.7.0, and the version in "environment.yml" is 1.7.1. Could this be the reason for the error?

**Comment by Clement25** [comment]
I am not sure but you can have a try. 
The fundamental problem is that 'torch.logdet' is not a very stable function. 
You may also check if set. other hyperparameters same value as in the paper.

**Comment by Xinghetayue** [comment]
> I am not sure but you can have a try. The fundamental problem is that 'torch.logdet' is not a very stable function. You may also check if set. other hyperparameters same value as in the paper.

ok，i will try
